### ISSUE TITLE
On DNS resolve fail, callback error is provided as response

### ISSUE THREAD

**Issue by yagop** [luvit/luvit#815]
```bash
$ luvit -e "dns=require('dns') dns.resolve4('fake.addr', p)"
nil	{ errcode = 3, errstr = 'name error' }
```

**Comment by rphillips** [comment]
Thank you for the report. I submitted a pull request, and we will try and get this fix released shortly.

**Comment by yagop** [comment]
Thanks a lot!

**Comment by rphillips** [comment]
released in 2.5.7
